FreeRTOS/FreeRTOS-Plus-TCP
# Fix the IP header checksum of the ICMP_ECHO_RESPONSE

Description
-----------

Since the IP header field usFragmentOffset is modified,
update the checksum in the IP header.
While I'm already there, simplify the computation of
the ICMP header checksum as well.

Test Steps
-----------
If ping request are sent with DONT FRAGMENT and
the response is sent without that bit, the IP header checksum is worng.
Linux ignores the ping response.

Ping works with this patch applied.


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
